### ISSUE TITLE
Live Preview "Host IP" setting does not support IPv6 addresses

### ISSUE THREAD

**Issue by austintraver** [microsoft/vscode-livepreview#179]
## Background

Support for specifying custom IP addresses was added by @andreamah as part of https://github.com/microsoft/vscode-livepreview/pull/117

The specific commit that adds this feature is 145f83606455f0b53f175f05142aa3a2d6632ea2 which has the following code snippet

https://github.com/microsoft/vscode-livepreview/blob/145f83606455f0b53f175f05142aa3a2d6632ea2/src/infoManagers/connectionManager.ts#L143-L149

## Proposal

I propose that this code is changed so that it uses the NodeJS standard library's [`net.isIP`](https://nodejs.org/dist/latest-v16.x/docs/api/net.html#netisipinput) which can properly validate IP addresses, and handle both IPv4 as well as IPv6

## Next Steps

I think this would be a fairly simple fix, which I'm happy to code this up myself and submit a pull-request with the corresponding changes. Are there any other places I'd need to change, aside from `connectionManager.ts`? Is using a `NodeJS` library possible in the context of this extension? Or would I have to use an API from the front-end JavaScript side of things?

**Comment by lszomoru** [comment]
@austintraver, good catch. Fixed it. I should be able to publish an update for the extension early next week.

**Comment by austintraver** [comment]
Wow, so soon? Way to go, thank you 🙏🏻 keep up the great work.

**Comment by lszomoru** [comment]
Updated version of the extension has been published to the marketplace.

**Comment by lszomoru** [comment]
@austintraver, can you please download the latest version of the extension and confirm that you can use IPv6? Thanks!

**Comment by austintraver** [comment]
Ah, @lszomoru unfortunately I'm still experiencing some trouble with version `v0.2.10`, but I definitely got further along than I did in the previous version!

Here is a brief demonstration of the current error I encounter, and a comparison between the behavior using the loopback address for IPv6 and IPv4 respectively:

https://user-images.githubusercontent.com/25112463/143992221-7191b8c2-b796-426d-b398-cdf4bd1e7017.mov

**Comment by lszomoru** [comment]
Thank you! Will take a look at this and will get it fixed.

**Comment by austintraver** [comment]
A word of caution: the issue is still marked "Closed"

**Comment by lszomoru** [comment]
@austintraver, I have looked into this today, and adding support for IPv6 will take more work than just adding support for an IPv6 address to the setting itself. I have published an updated version (`0.2.11`) of the extension that reverts the change to support an IPv6 for the setting in order to avoid confusion. I will further look into this in the month of December. Sorry about the delay.

**Comment by austintraver** [comment]
> @austintraver, I have looked into this today, and adding support for IPv6 will take more work than just adding support for an IPv6 address to the setting itself. I have published an updated version (`0.2.11`) of the extension that reverts the change to support an IPv6 for the setting in order to avoid confusion. I will further look into this in the month of December. Sorry about the delay.

No problem. Documenting that this feature is not (yet) supported is a perfectly acceptable response. Thank you for giving this issue the attention that it received. Keep up the great work.
